python/housinginsights/ingestion/DataReader.py

```diff
@@ -363,3 +456,57 @@ def _verify_required_metadata_columns_exist_in_actual(self, field_list):
                                       'data'.format(field['source_name']))
                 success = False
         return success
+
+    def _contains_leading_bom(self, str, encoding='utf-8'):
+        """
+        Returns True if string starts with 0xEFBBBF byte order mark
+        seuence and False otherwise.
+
+        :param str: the string to evaluate
+        :type str: str
+
+        :param encoding: str encoding (default: 'utf-8'; 
+                                                'latin-1' also supported)
+        :type encoding: str
+        """
+        if (encoding == 'utf-8'):
+            bom_as_hex = "efbbbf"   # hex encoding of utf-8 BOM
+            str_as_hex = "".join("{:02x}".format(c) for c in str.encode())
+            if str_as_hex.startswith(bom_as_hex):
+                return True
+        elif (encoding == 'latin-1'):
+            bom_in_latin_1 = "ï»¿"  # latin-1-encoded version of BOM
```

Comment:
This should work now; DataReader now consistently passes the encoding through to all encoding-sensitive methods.